src/client/QXmppMovedManager.cpp

```diff
@@ -0,0 +1,260 @@
+                                           });
+}
+
+///
+/// Notify a contact we have moved
+///
+/// \param contactBareJid JID of the contact to send the subscription request
+/// \param oldBareJid JID of the old account we moved from
+/// \param reason The reason of the move
+///
+/// \return the result of the action
+///
+QXmppTask<QXmpp::SendResult> QXmppMovedManager::notifyContact(const QString &contactBareJid, const QString &oldBareJid, const QString &reason)
+{
+    QXmppPresence packet;
+    packet.setTo(QXmppUtils::jidToBareJid(contactBareJid));
+    packet.setType(QXmppPresence::Subscribe);
+    packet.setStatusText(reason);
+    packet.setOldJid(oldBareJid);
+    return client()->sendSensitive(std::move(packet));
+}
+
+/// \cond
+void QXmppMovedManager::onRegistered(QXmppClient *client)
+{
+    connect(client, &QXmppClient::connected, this, [this, client]() {
+        if (client->streamManagementState() == QXmppClient::NewStream) {
+            resetCachedData();
+        }
+    });
+
+    d->discoveryManager = client->findExtension<QXmppDiscoveryManager>();
+    Q_ASSERT_X(d->discoveryManager, "QXmppMovedManager", "QXmppDiscoveryManager is missing");
+
+    connect(d->discoveryManager, &QXmppDiscoveryManager::infoReceived, this, &QXmppMovedManager::handleDiscoInfo);
+
+    d->pubSubManager = client->findExtension<QXmppPubSubManager>();
+    Q_ASSERT_X(d->pubSubManager, "QXmppMovedManager", "QXmppPubSubManager is missing");
+}
+
+void QXmppMovedManager::onUnregistered(QXmppClient *client)
+{
+    disconnect(d->discoveryManager, &QXmppDiscoveryManager::infoReceived, this, &QXmppMovedManager::handleDiscoInfo);
+    resetCachedData();
+    disconnect(client, &QXmppClient::connected, this, nullptr);
+}
+/// \endcond
+
+///
+/// Handles incoming service infos specified by \xep{0030, Service Discovery}
+///
+/// \param iq received Service Discovery IQ stanza
+///
+void QXmppMovedManager::handleDiscoInfo(const QXmppDiscoveryIq &iq)
+{
+    // Check the server's functionality to support MOVED feature.
+    if (iq.from().isEmpty() || iq.from() == client()->configuration().domain()) {
+        // Check whether MOVED is supported.
+        setSupportedByServer(iq.features().contains(ns_moved));
+    }
+}
+
+///
+/// Ensure both JIDs match
+///
+/// \param newBareJid JID of the contact that sent the subscription request
+/// \param pepBareJid JID of the new account as fetched from the old account statement
+///
+/// \return the result of the action
+///
+QXmppClient::EmptyResult QXmppMovedManager::movedJidsMatch(const QString &newBareJid, const QString &pepBareJid) const
+{
+    if (newBareJid == pepBareJid) {
+        return QXmpp::Success {};
+    }
+
+    return QXmppError { QStringLiteral("The JID does not match user statement."), {} };
+}
+
+///
+/// Sets whether the own server supports \xep{0283, Moved}.
+///
+/// \param supportedByServer whether \xep{0283, Moved} is supported by the own server
```

Comment:
I removed that from the MIX manager as well because it is clear that the own server is meant.
```suggestion
/// \param supportedByServer whether \xep{0283, Moved} is supported by the server
```